openmdao/core/system.py

```diff
@@ -1527,9 +1527,11 @@ def _setup_global_shapes(self):
                         high_size = np.prod(high_dims)
                         dim1 = global_size // high_size
                         if global_size % high_size != 0:
-                            raise RuntimeError("%s: Global size of output '%s' (%s) does not agree "
-                                               "with local shape %s" % (self.msginfo, abs_name,
-                                                                        global_size, local_shape))
+                            raise RuntimeError(f"{self.msginfo}: All but the first dimension of "
+                                f"the shape's local parts in a distributed variable must match "
+                                f"across processes. For output '{abs_name}', local shape "
+                                f"{local_shape} in MPI rank {MPI.COMM_WORLD.rank} has a higher "
+                                "dimension that differs in another rank.")
```

Comment:
there should be test for this...

Reply:
In the process of writing the test, I found that `_setup_global_shapes()` would not raise errors in all ranks. Since the exception had to be caught for the test, it would hang waiting for the other process instead of aborting. I added the `_comm_global_shape_status()` function so that all ranks would know whether the global shape was valid and raise an exception if not.